src/main/java/com/internet/shop/dao/jdbc/ShoppingCartDaoJdbcImpl.java

```diff
@@ -0,0 +1,207 @@
+                + " = shopping_carts_products.shopping_carts_id"
+                + " LEFT JOIN products ON products.product_id"
+                + " = shopping_carts_products.products_id"
+                + " where shopping_carts.user_id = ?";
+
+        try (Connection connection = DbConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, userId);
+            ResultSet sqlResult = preparedStatement.executeQuery();
+            if (sqlResult.next()) {
+                ShoppingCart cart = parseResult(sqlResult);
+                return Optional.of(cart);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get cart by userId = " + userId, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public ShoppingCart create(ShoppingCart cart) {
+        String query = "INSERT INTO shopping_carts (user_id)"
+                + " VALUES (?)";
+
+        try (Connection connection = DbConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement =
+                    connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            preparedStatement.setLong(1, cart.getUserId());
+            preparedStatement.executeUpdate();
+            ResultSet resultKey = preparedStatement.getGeneratedKeys();
+            if (resultKey.next()) {
+                cart.setId(resultKey.getLong(1));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("ShoppingCart for user with id " + cart.getUserId()
+                    + " has not been added", e);
+        }
+        return cart;
+    }
+
+    @Override
+    public Optional<ShoppingCart> get(Long id) {
+        String query = "SELECT * FROM shopping_carts"
+                + " JOIN shopping_carts_products ON shopping_carts.shopping_cart_id"
+                + " = shopping_carts_products.shopping_carts_id"
+                + " JOIN products ON products.product_id"
+                + " = shopping_carts_products.products_id"
+                + " where shopping_carts.shopping_cart_id = ? ";
+
+        try (Connection connection = DbConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, id);
+            ResultSet sqlResult = preparedStatement.executeQuery();
+            if (sqlResult.next()) {
+                ShoppingCart cart = parseResult(sqlResult);
+                return Optional.of(cart);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get shopping cart by id = " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    private ShoppingCart parseResult(ResultSet sqlResult) throws SQLException {
```

Comment:
rename, please to parseShopCart or similar...

Reply:
Ok. Done